aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1701,6 +1749,41 @@ impl VMAdapter for AptosVM {
             },
         })
     }
+
+    fn execute_single_transaction_sequential(
+        &self,
+        txn: &PreprocessedTransaction,
+        view: &impl StateView,
+        log_context: &AdapterLogSchema,
+    ) -> Result<(VMStatus, VMOutput, Option<String>), VMStatus> {
+        match self.execute_single_transaction(txn, &view.as_move_resolver(), log_context, true) {
+            Ok((vm_status, vm_output, sender)) => match vm_output.try_materialize(view) {
+                Ok(vm_output) => {
+                    return Ok((vm_status, vm_output, sender));
+                },
+                Err(vm_status) => {
```

Comment:
Seems like quite a bit of code is duplicated here? Maybe @zekun000 or @gelash know more Rusty ways of writing this

Reply:
I agree. If anyone has ideas on reducing the duplication, I'm happy to change it.

Reply:
Some ideas:
What about using map_err and map here? We can map non-aggregator-error to Some(error) and otherwise to None and have a single match instead.

Reply:
Tried it after your comment. Didn't simplify things. Now, we need to selectively return Ok(..) or Err(..) statements in the match statements. If there is an aggregator error, we need to re-execute transaction later on. It didn't simply the code when using map_err and map.

Reply:
So what I meant was something like:
```
// returns none if it is an aggregator error, otherwise Some(VMStatus).
let clear_aggregator_error = |status: VMStatus| -> Option<VMStatus> { .. };
 
let result = self
  .execute_single_transaction(txn, &view, log_context, true)
  .map_err(|status| clear_aggregator_error(status))
  .map(|(status, output, sender)| {
    // If execution succeeds, try to apply deltas.
    output.try_materialize(view)
       .map(|output| (status, output, sender))
       .map_err(|status| clear_aggregator_error(status))
   });

// Now we just re-execute if Ok(Err(None)) or Err(None)
```
I think this is a more idiomatic Rust way (cc @gelash @zekun000 again), but if you think this is worse, let's keep it as is now. 

On the other note, if aggregator fails in ` self.execute_single_transaction(txn, &view, log_context, true)`, why do we need this re-execution? it can happen if you have an aggregator in Data state or do `aggregator::add(&agg, 2 * limit))` which clearly overflows. Maybe I am missing something?